src/engine/strat_engine/blockdevmgr.rs

```diff
@@ -40,17 +42,50 @@ pub fn resolve_devices(paths: &[&Path]) -> io::Result<HashSet<Device>> {
         .collect()
 }
 
+#[derive(Debug)]
+pub struct BlkDevSegment {
+    pub blockdev: Rc<RefCell<BlockDev>>,
+    pub start: Sectors,
+    pub length: Sectors,
+}
+
+impl BlkDevSegment {
+    pub fn new(blockdev: Rc<RefCell<BlockDev>>, start: Sectors, length: Sectors) -> BlkDevSegment {
+        BlkDevSegment {
+            blockdev,
+            start,
+            length,
+        }
+    }
+
+    pub fn to_segment(&self) -> Segment {
+        Segment::new(*self.blockdev.borrow().device(), self.start, self.length)
+    }
+}
+
+/// Build a Vec<Segment> from BlkDevSegments. This is useful for calls
+/// to the devicemapper library.
+pub fn convert_to_dm_segs(bsegs: &[BlkDevSegment]) -> Vec<Segment> {
+    bsegs
+        .iter()
+        .map(|bseg| bseg.to_segment())
+        .collect::<Vec<_>>()
+}
+
```

Comment:
Proposed fun

```
fn inject_bd(bd: Rc<RefCell<BlockDev>>, segs: &[(Sectors, Sectors)]) -> Vec<BlkDevSegment> {
     segs.into_iter().map(|(start, length)| { BlkDevSegment::new(Rc::clone(bd), start, length) }))
}
```
used below.

Reply:
If this was used in more than one spot I'd be in favor, but as it is I don't think a one-line function is quite justified.

However, maybe it's clearer if we have a let binding that does the map, and then pass that to extend() instead of a single big expression? Let's try that.

Reply:
It could be used three times in test code, which is a hint.